Version Control on Tables

TanuMalik opened the pull request.

WillEngler commented:
I pulled in the history tracking script and added it to all of our point tables. Unfortunately, it didn't fare well in production.

Check out the ETL log in the admin panel starting at January 22, 2016 13:46 for all the details, but here's one example:

```
PlenarioETLError: ProgrammingError('(psycopg2.ProgrammingError) syntax error at or near "s"\nLINE 1: ...Post name, address, and telephone of owner, owner\'s agent fo...\n ^\nQUERY: INSERT INTO audit.micro_market_recovery_program_violations_and_inspe_history VALUES (\'2016-01-22 15:14:13.024878+00\',\'D\',\'(BLDG,BLDG_PERM,"PERMIT - NEW CONSTRUCTION",1418795,1416655,"2006-08-15 00:00:00",BL00812,CN_ANNUAL,f,2,f,,FAILED,1291458,"2009-09-23 17:05:36","2009-09-23 17:05:36","Building, post management sign",CN196029,"SEQ #: 6","2005-09-02 00:00:00",,,6,PCR31765,"POST OWNER/MANAGERS NAME/#","Post name, address, and telephone of owner, owner\'s agent for managing, controlling or collecting rents, and any other person managing or controlling building conspicuously where accessible or visible to public way. (13-12-030)",Austin,5500,W,LAKE,ST,,60644,25,28,1139449.8818507,1902230.53555938,41.8878297345816,-87.7633533997474,,81047,2556,"(41.88782973458163, -87.76335339974737)",c1e4f281cf7b7f51b0325c3358d03f95,0101000020E610000027CD37C8DAF055C0E0389D67A4F14440,"2005-09-02 00:00:00")\')\nCONTEXT: PL/pgSQL function if_modified() line 47 at EXECUTE statement\n',)
```
I removed the trigger from the tables and removed it from the ETL module for now.

TanuMalik commented:
Hi Will

Can you give me some clues how to debug on production system so I am careful.

The error is not informative so can I take the export this table out.

Let me know if you have any better ideas.

Tanu



On Jan 23, 2016, at 10:38 AM, Will Engler <notifications@github.com<mailto:notifications@github.com>> wrote:


I pulled in the history tracking script and added it to all of our point tables. Unfortunately, it didn't fare well in production.

Check out the ETL log in the admin panel starting at January 22, 2016 13:46 for all the details, but here's one example:

PlenarioETLError: ProgrammingError('(psycopg2.ProgrammingError) syntax error at or near "s"\nLINE 1: ...Post name, address, and telephone of owner, owner\'s agent fo...\n ^\nQUERY: INSERT INTO audit.micro_market_recovery_program_violations_and_inspe_history VALUES (\'2016-01-22 15:14:13.024878+00\',\'D\',\'(BLDG,BLDG_PERM,"PERMIT - NEW CONSTRUCTION",1418795,1416655,"2006-08-15 00:00:00",BL00812,CN_ANNUAL,f,2,f,,FAILED,1291458,"2009-09-23 17:05:36","2009-09-23 17:05:36","Building, post management sign",CN196029,"SEQ #: 6","2005-09-02 00:00:00",,,6,PCR31765,"POST OWNER/MANAGERS NAME/#","Post name, address, and telephone of owner, owner\'s agent for managing, controlling or collecting rents, and any other person managing or controlling building conspicuously where accessible or visible to public way. (13-12-030)",Austin,5500,W,LAKE,ST,,60644,25,28,1139449.8818507,1902230.53555938,41.8878297345816,-87.7633533997474,,81047,2556,"(41.88782973458163, -87.76335339974737)",
 c1e4f281
 cf7b7f51b0325c3358d03f95,0101000020E610000027CD37C8DAF055C0E0389D67A4F14440,"2005-09-02 00:00:00")\')\nCONTEXT: PL/pgSQL function if_modified() line 47 at EXECUTE statement\n',)


I removed the trigger from the tables and removed it from the ETL module for now.

—
Reply to this email directly or view it on GitHub<https://github.com/UrbanCCD-UChicago/plenario/pull/209#issuecomment-174198691>.
